x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/TrainedModelStatsService.java

```diff
@@ -47,12 +48,16 @@
     private static final Logger logger = LogManager.getLogger(TrainedModelStatsService.class);
     private static final TimeValue PERSISTENCE_INTERVAL = TimeValue.timeValueSeconds(1);
 
+    private static final String STATS_UPDATE_SCRIPT_TEMPLATE = "" +
+        "    ctx._source.{0} += params.{0};\n" +
+        "    ctx._source.{1} += params.{1};\n" +
+        "    ctx._source.{2} += params.{2};\n" +
+        "    ctx._source.{3} = params.{3};";
     // Script to only update if stats have increased since last persistence
-    private static final String STATS_UPDATE_SCRIPT = "" +
-        "    ctx._source.missing_all_fields_count += params.missing_all_fields_count;\n" +
-        "    ctx._source.inference_count += params.inference_count;\n" +
-        "    ctx._source.failure_count += params.failure_count;\n" +
-        "    ctx._source.time_stamp = params.time_stamp;";
+    private static final String STATS_UPDATE_SCRIPT = Messages.getMessage(STATS_UPDATE_SCRIPT_TEMPLATE,
+        InferenceStats.INFERENCE_COUNT.getPreferredName(),
```

Comment:
Are we missing `InferenceStats.MISSING_ALL_FIELDS_COUNT` here?

Reply:
yep, I am :/